File: app/src/main/java/com/example/imageviewer/ImageListAdapter.java
package com.example.imageviewer;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

public class ImageListAdapter extends RecyclerView.Adapter<ImageListAdapter.ImageDataViewHolder>{

    ArrayList<ImageData> imageData;

    Context context;

    public ImageListAdapter(ArrayList<ImageData> imageData){
        this.imageData = imageData;
    }

    class ImageDataViewHolder extends RecyclerView.ViewHolder{

        TextView imageNameView;
        ImageView imageView;
        View parentView;


        public ImageDataViewHolder(@NonNull View itemView) {
            super(itemView);
            Log.i("testone", "onBindViewHolder : called,");
            imageNameView = itemView.findViewById(R.id.image_name_text_view);
            imageView = itemView.findViewById(R.id.imageView);
            parentView = itemView.findViewById(R.id.list_parent);
        }
    }


    @NonNull
    @Override
    public ImageDataViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int i) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.image_item_layout, parent, false);
        Log.i("testone", "onBindViewHolder : called,");
        return new ImageDataViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ImageDataViewHolder viewHolder, int i) {
        final ImageData data = imageData.get(i);

        Log.i("testone", "onBindViewHolder : called,");

        viewHolder.imageView.setImageURI(Uri.parse(data.getUri()));
        viewHolder.imageNameView.setText(data.getName());
        viewHolder.imageView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(v.getContext(), ImageDetails.class);
                intent.putExtra(ImageData.TAG, imageData);
                ((Activity)v.getContext()).startActivityForResult(intent,MainActivity.TO_DETAILS_REQUEST);
            }
        });

    }

    @Override
    public int getItemCount() {
        return this.imageData.size();
    }
}

File: app/src/main/java/com/example/imageviewer/ImageListAdapter.kt
package com.example.imageviewer

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat.startActivity
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.image_item_layout.view.*

class ImageListAdapter (val data: ArrayList<ImageData>) :
    RecyclerView.Adapter<ImageListAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val image: ImageView = view.image_view
        val imageName: TextView = view.image_name
        val imageUri: TextView = view.image_uri

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val viewGroup = LayoutInflater.from(parent.context).inflate(R.layout.image_item_layout, parent, false)
        return ViewHolder(viewGroup)
    }

    override fun getItemCount(): Int {
        return data.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.image.setImageURI(data[position].fileUri)
        holder.imageName.text=  data[position].fileName
        holder.imageUri.text= data[position].fileUriString
        //holder.itemView.Set



    }
}

